src/shared/components/post/post-listing.tsx

```diff
@@ -1723,6 +1725,16 @@ export class PostListing extends Component<PostListingProps, PostListingState> {
     event.preventDefault();
     i.setState({ imageExpanded: !i.state.imageExpanded });
     setupTippy();
+
+    const auth = myAuth();
+    if (auth && !i.props.post_view.read) {
+      const form: MarkPostAsRead = {
+        post_id: i.props.post_view.post.id,
+        read: true,
+        auth: auth,
+      };
+      HttpService.client.markPostAsRead(form);
```

Comment:
Don't do HTTPService calls here. Follow the convention with the rest of the actions, and add `onMarkPostAsRead` to the props.

Check out `onSavePost` for how that works.

Reply:
As a professional software engineer it smells weird to me when I need to duplicate so much code to add a new feature (+56 lines changed), while the original had no duplicate code (+13 lines changed) and didn't need the reader to know 10 different files.

If different places in the code want to do something when the post is marked as read, they should add an event listener on the "read" state of the post (or a more general state change), in my opinion.

I am used to programming using the Qooxdoo framework, where you have a domain object (something like Post in this case), which would have properties like "read: Boolean". Whenever you click to expand the image, you would do the HTTP request, and if that succeeded, you do a `setRead(true)`, and Qooxdoo would fire all event listeners that are listening to that property of that object. So if for example the Post is shown on the home page, the home page code would `post.addListener('changeRead', this.onChangeRead)` to register the listener, so that any changes to that property causes Home#onChangeRead() to be called.

Nonetheless, I have made a commit that does what you asked.